packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -83,6 +83,35 @@ export async function commitPromptAsync(
   });
 }
 
+export async function makeProjectMetadataFileAsync(archivePath: string): Promise<{ path: string; size: number }> {
+  const spinner = ora('Creating project metadata file');
+  const timerLabel = 'makeProjectMetadataFileAsync';
+  startTimer(timerLabel);
+
+  const metadataLocation = path.join(getTmpDirectory(), `${uuidv4()}.json`);
+  const archiveContent: string[] = [];
+
+  await tar.list({
+    file: archivePath,
+    onentry: (entry: tar.ReadEntry) => {
+      if (entry.type === 'File') {
+        archiveContent.push(entry.path);
+      }
+    }
+  });
+
+  await fs.writeJSON(metadataLocation, {
+    archiveContent
+  });
```

Comment:
What does the example metadata file look like?

Reply:
```
{
    "archiveContent": [
        "paths/to/files"
    ]
}
```